pantheon-bundle/src/main/java/com/redhat/pantheon/servlet/DocumentVariantRenderServlet.java

```diff
@@ -0,0 +1,119 @@
+package com.redhat.pantheon.servlet;
+
+import com.redhat.pantheon.asciidoctor.AsciidoctorService;
+import com.redhat.pantheon.helper.PantheonConstants;
+import com.redhat.pantheon.model.document.DocumentVariant;
+import com.redhat.pantheon.model.document.DocumentVersion;
+import org.apache.commons.lang3.LocaleUtils;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.servlets.SlingSafeMethodsServlet;
+import org.apache.sling.servlets.annotations.SlingServletResourceTypes;
+import org.osgi.framework.Constants;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.Servlet;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.Writer;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+
+import static com.redhat.pantheon.servlet.ServletUtils.paramValueAsBoolean;
+
+/**
+ * Renders an HTML preview for a single document variant.
+ * To provide parameters to the asciidoc generation process, provide the parameters with their name prefixed
+ * with "ctx_".
+ *
+ * For example, if an asciidoc attribute of name 'product' needs to be passed, there will need to be a
+ * query parameter of name 'ctx_product' provided in the url.
+ */
+@Component(
+        service = Servlet.class,
+        property = {
+                Constants.SERVICE_DESCRIPTION + "=Servlet which transforms asciidoc content into html",
+                Constants.SERVICE_VENDOR + "=Red Hat Content Tooling team"
+        })
+@SlingServletResourceTypes(
+        resourceTypes = { "pantheon/moduleVariant", "pantheon/assemblyVariant" },
+        methods = "GET",
+        extensions = "preview")
+public class DocumentVariantRenderServlet extends SlingSafeMethodsServlet {
+
+    private static final Set<String> RELEASED_SUFFIXES = new HashSet<>();
+    static {
+        RELEASED_SUFFIXES.add("/released");
+        RELEASED_SUFFIXES.add("/");
+        RELEASED_SUFFIXES.add("");
+        RELEASED_SUFFIXES.add(null);
+    }
```

Comment:
are those allowed values for RELEASED_SUFFICES?

Reply:
Yes, the idea here is that the suffix is optional. If you don't specify it, we assume you want the released version. This covers all the cases: [document, document/, document/released]